core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -278,8 +278,8 @@ public String getPolicyName() {
   private ObjectPool<Executor> balancerRpcExecutorPool;
   @Mock
   private CallCredentials creds;
-  @Mock
-  private Executor offloadExecutor;
+  private FakeClock offloadFakeClock = new FakeClock();
```

Comment:
Earlier I had thought the CallCredentials on line 2349 were using the executor. But since it isn't it means we can actually keep this as a Mock.

It'd be fine to keep this an a FakeClock, but to avoid deleting the lines in nameResolverArgsPropagation the equivalent would be:
```java
    assertEquals(0, offloadFakeClock.numPendingTasks());
    args.getOffloadExecutor()
        .execute(
            new Runnable() {
              @Override
              public void run() {}
            });
    assertEquals(1, offloadFakeClock.numPendingTasks());
```

Reply:
I thought that the CallCredentials is using the executor as otherwise the pending task wouldn't be added to the offloadExecutor, which is now the offloadFakeClock's ScheduledExecutorService.  If there were no association between offloadFakeClock and offloadExecutor, how would one affect the other?